src/angular/form-elements/input/input.component.html.ts

```diff
@@ -1,19 +1,19 @@
 export default `
-<div class="sdc-input">
+<div class="sdc-input ">
     <label class="sdc-input__label" *ngIf="label" [ngClass]="{'required':required}">{{label}}</label>
     <input
-        class="sdc-input__input"
+        class="sdc-input__input {{classNames}}"
         [ngClass]="{'error': !valid, 'disabled':disabled}"
-        type="text"
         [name]="name"
         [placeholder]="placeHolder"
         [(ngModel)]="value"
         [maxlength]="maxLength"
         [minlength]="minLength"
-        [type]="type"
+        [type]="type || 'text'"
```

Comment:
It might be better to give an init value for "type" property in the Component class.